### ISSUE TITLE
Compiler specs fail on build with Crystal 1.1.0

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#10961]
Building the compiler specs with a Crystal 1.1.0 compiler results in an invalid memory access on multiple platforms (see https://github.com/crystal-lang/crystal/pull/10955/checks?sha=84909745c5247e3ec6532b450d9cdd3a4397dce4).

This spec produces the error in isolation:
```console
$ bin/crystal spec spec/compiler/codegen/proc_spec.cr -e 9823
Invalid memory access (signal 11) at address 0x0
[0x8855c6] *Exception::CallStack::print_backtrace:(Int32 | Nil) +118
[0x86ebfa] ~procProc(Int32, Pointer(LibC::SiginfoT), Pointer(Void), Nil) +330
[0x7f66da5c73c0] ???
[0x7f66db29ec0c] _ZN4llvm22report_bad_alloc_errorEPKcb +108
[0x7f66db2e24af] _ZN4llvm15SmallVectorBase8grow_podEPvmm +47
[0x7f66db323b71] _ZN4llvm19raw_svector_ostream10write_implEPKcm +49
[0x7f66db3225dd] _ZN4llvm11raw_ostream5writeEPKcm +429
[0x7f66db2ff776] _ZNK4llvm5Twine5printERNS_11raw_ostreamE +22
[0x7f66db29e9fd] _ZN4llvm18report_fatal_errorERKNS_5TwineEb +333
[0x7f66dc593503] _ZN4llvm5MCJIT25getPointerToNamedFunctionENS_9StringRefEb +515
[0x0] ???
$ bin/crystal --version
Crystal 1.1.0 [af095d72d] (2021-07-14)

LLVM: 10.0.1
Default target: x86_64-unknown-linux-gnu
$ git describe
1.1.0
```

**Comment by asterite** [comment]
Does the program in the spec fails too? That is, this program:

```crystal
      abstract struct Parent
        abstract def work(a : Int32, b : Int32)

        def get
          ->work(Int32, Int32)
        end
      end

      struct Child1 < Parent
        def work(a : Int32, b : Int32)
          a &+ b
        end
      end

      struct Child2 < Parent
        def work(a : Int32, b : Int32)
          a &- b
        end
      end

      Child1.new.as(Parent).get

```

**Comment by straight-shoota** [comment]
@asterite The compiler binary itself does not error on the above program. I even tried with a 3rd generation 1.1.0 compiler just to make sure.
It looks like it's only a spec error related to the LLVM JIT compiler. Deactivating JIT compilation for that spec makes it pass.

So this error should have no effect on any actual use of the Crystal compiler. It only appears in the compiler spec suite and not in the compiler itself.

**Comment by asterite** [comment]
Cool! We avoid using the JIT in many places because some things don't work there. I don't know why, if it's a bug or it's just un finished in the JIT. So maybe we can require prelude in that spec for now.

**Comment by asterite** [comment]
(requiring prelude tells the spec runner not to use the JIT)

**Comment by straight-shoota** [comment]
Resolved in  #10964
